fvm/transactionInvoker.go

```diff
@@ -135,15 +134,88 @@ func (executor *transactionExecutor) Execute() error {
 	return nil
 }
 
+func (executor *transactionExecutor) Preprocess() error {
+	if !executor.TransactionBodyExecutionEnabled {
+		return nil
+	}
+
+	err := executor.PreprocessTransactionBody()
+	return executor.handleError(err, "preprocessing")
+}
+
+func (executor *transactionExecutor) Execute() error {
+	return executor.handleError(executor.execute(), "executing")
+}
+
+// PreprocessTransactionBody preprocess parts of a transaction body that are
+// infrequently modified and are expensive to compute.  For now this includes
+// reading meter parameter overrides and parsing programs.
+func (executor *transactionExecutor) PreprocessTransactionBody() error {
+	meterParams, err := getBodyMeterParameters(
+		executor.ctx,
+		executor.proc,
+		executor.txnState,
+		executor.derivedTxnData)
+	if err != nil {
+		return fmt.Errorf("error gettng meter parameters: %w", err)
+	}
+
+	txnId, err := executor.txnState.BeginNestedTransactionWithMeterParams(
+		meterParams)
+	if err != nil {
+		return err
+	}
+	executor.nestedTxnId = txnId
+
+	executor.txnBodyExecutor = executor.cadenceRuntime.NewTransactionExecutor(
+		runtime.Script{
+			Source:    executor.proc.Transaction.Script,
+			Arguments: executor.proc.Transaction.Arguments,
+		},
+		common.TransactionLocation(executor.proc.ID))
+
+	// This initializes various cadence variables and parses the programs used
+	// by the transaction body.
+	err = executor.txnBodyExecutor.Preprocess()
+	if err != nil {
+		executor.errs.Collect(
+			fmt.Errorf(
+				"transaction preprocess failed: %w",
+				err))
+
+		// We shouldn't early exit on non-failure since we need to deduct fees.
+		if executor.errs.CollectedFailure() {
+			return executor.errs.ErrorOrNil()
+		}
+
+		// NOTE: We need to restart the nested transaction in order to pause
+		// for fees deduction.
+		err = executor.txnState.RestartNestedTransaction(txnId)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Pause the transaction body's nested transaction in order to interleave
+	// auth and seq num checks.
+	pausedState, err := executor.txnState.Pause(txnId)
+	if err != nil {
+		return err
+	}
+	executor.pausedState = pausedState
+
+	return nil
+}
+
 func (executor *transactionExecutor) execute() error {
 	if executor.AuthorizationChecksEnabled {
 		err := executor.CheckAuthorization(
 			executor.ctx.Tracer,
 			executor.proc,
 			executor.txnState,
 			executor.AccountKeyWeightThreshold)
-		if err != nil {
-			return err
+		if executor.errs.Collect(err).CollectedFailure() {
```

Comment:
I finally figured out the issue with this PR.  This broke verification test because this used to early exit when auth check / seq num check fail on user error.

We must unconditionally resume the paused transaction body nested transaction, and commit and it, or else we may end up with missing registers on the execution result.  This then causes verification failure.